test/generate-xml.sh

```diff
@@ -78,10 +86,22 @@ for hexfile in "$directory"/*.hex;  do
              ;;
         *)
              # no old -> rename XML
-             echo "Create $filename.xml"
-             mv "$directory/$filename.xml.new" "$directory/$filename.xml"
+             echo "Create $filename$suffix"
+             mv "$directory/$filename$suffix.new" "$directory/$filename$suffix"
              rm "$directory/$filename.dif"
              ;;
     esac
+
+    return 0
+}
+
+for hexfile in "$directory"/*.hex;  do
+    if [ ! -f "$hexfile" ]; then
+        continue
+    fi
+
+    generate_xml "$directory" "$hexfile" "default"
+
+    generate_xml "$directory" "$hexfile" "normalized"
```

Comment:
Here we don't seem to look at the return value of the function

Reply:
Or did you intend me to update the commit "make build stop on broken tests"?

Reply:
It was your idea, so you should get the credits for it.